src/client/renderer/r_lighting.c

```diff
@@ -409,7 +423,9 @@ static void R_UpdateShadows(r_lighting_t *l) {
 	}
 
 	// otherwise, refresh all shadow information based on the new illuminations
-	memset(l->shadows, 0, sizeof(l->shadows));
+	for (size_t i = 0; i < lengthof(l->shadows); i++) {
+		R_FreeShadow(&l->shadows[i]);
+	}
```

Comment:
Instead of blindly wiping all shadows, we iterate them, and decrement the shadow count of any impacted planes. This should be safe. I can't conceive of how a shadow could hold an invalid reference to a plane -- at least not on the first map load. I don't get it.